static/app/components/charts/eventsChart.tsx

```diff
@@ -223,12 +224,16 @@ class Chart extends React.Component<ChartProps, State> {
     if (previousSeriesTransformer) {
       previousSeries = previousSeriesTransformer(previousTimeseriesData);
     }
+    const chartColors = timeseriesData.length
+      ? colors?.slice(0, series.length) ?? [
+          ...theme.charts.getColorPalette(timeseriesData.length - 2),
+        ]
+      : undefined;
+    if (chartColors && chartColors.length && hasOther) {
+      chartColors[chartColors.length - 1] = theme.charts.other;
+    }
```

Comment:
I think what this is doing right now is picking up an extra colour and replacing it with grey:

`topEvents` is `5` and `timeseriesData.length` is `6`, `theme.charts.getColorPalette(timeseriesData.length - 2)` will return an array with 6 colours. We then replace the last colour with grey.

Don't think this is a big issue but a small improvement could be if we know we're going to replace the last colour with grey, we can just fetch 5 colours instead and append grey as the 6th element. ie `theme.charts.getColorPalette(timeseriesData.length - 3)`. This should result in a _slight_ improvement to colour range. 

We might also have to update the results table colours as well if we make this change since the table probably doesn't access colours from the array here.